setup.py

```diff
@@ -182,23 +140,37 @@ def members_appended(*ds):
         print('Warning: can not import pkgconfig python package.')
         pc = None
 
+    def lib_ext_kwargs(pc, prefix_env_var, lib_name, lib_pkg_name, pc_version):
+        system_prefix = os.environ.get(prefix_env_var)
+        if system_prefix:
+            print(f'Detected and preferring {lib_pkg_name} [via {prefix_env_var}]')
+            return dict(include_dirs=[os.path.join(system_prefix, 'include')],
+                        library_dirs=[os.path.join(system_prefix, 'lib')],
+                        libraries=[lib_name])
+
+        if pc and pc.installed(lib_pkg_name, pc_version):
+            print(f'Detected and preferring {lib_pkg_name} [via pkg-config]')
+            return pc.parse(lib_pkg_name)
+        raise Exception(f'Could not find {lib_name} lib/headers, please set {prefix_env_var}')
+
+
     crypto_ext_kwargs = members_appended(
         dict(sources=[crypto_ll_source, crypto_helpers]),
-        setup_crypto.crypto_ext_kwargs(pc, system_prefix_openssl),
+        lib_ext_kwargs(pc, 'BORG_OPENSSL_PREFIX', 'crypto', 'libcrypto', '>=1.1.1'),
         dict(extra_compile_args=cflags),
     )
 
     compress_ext_kwargs = members_appended(
         dict(sources=[compress_source]),
-        setup_compress.lz4_ext_kwargs(pc, system_prefix_liblz4),
-        setup_compress.zstd_ext_kwargs(pc, system_prefix_libzstd),
+        lib_ext_kwargs(pc, 'BORG_LIBLZ4_PREFIX', 'lz4', 'liblz4', '>= 1.7.0'),
+        lib_ext_kwargs(pc, 'BORG_LIBZSTD_PREFIX', 'zstd', 'libzstd', '>= 1.3.0'),
         dict(extra_compile_args=cflags),
     )
 
     checksums_ext_kwargs = members_appended(
         dict(sources=[checksums_source]),
-        setup_checksums.xxhash_ext_kwargs(pc, system_prefix_libxxhash),
-        setup_checksums.deflate_ext_kwargs(pc, system_prefix_libdeflate),
+        lib_ext_kwargs(pc, 'BORG_LIBXXHASH_PREFIX', 'xxhash', 'libxxhash', '>= 0.7.3'),
```

Comment:
0.8.1 was usually required for pkg-config support.

Reply:
I got some test errors with 0.8.1. Currently it's 0.7.3 in code, but 0.8.1 in the comment. Will try to use 0.8.1 everywhere.

https://github.com/borgbackup/borg/blob/master/setup_checksums.py#L13

Reply:
0.8.1 fails in CI. Maybe their version is older. Will keep 0.7.3 for now.

```
Traceback (most recent call last):
  File "/home/runner/work/borg/borg/setup.py", line 177, in <module>
    lib_ext_kwargs(pc, 'BORG_LIBXXHASH_PREFIX', 'xxhash', 'libxxhash', '>= 0.8.1'),
...
Exception: Could not find xxhash lib/headers, please set BORG_LIBXXHASH_PREFIX
Error: Process completed with exit code 1.
```

Reply:
ah right, it should of course work on the CI we have now.